client/gutenberg/extensions/markdown/jetpack-markdown-block-editor.jsx

```diff
@@ -82,37 +77,19 @@ class JetpackMarkdownBlockEditor extends Component {
 			}
 		};
 
-		// Manages css classes for each panel based on component's state
-		const classesForPanel = function( panelName ) {
-			return classNames( {
-				'components-tab-button': true,
-				'is-active': this.state.activePanel === panelName,
-				[ `${ className }__${ panelName }-button` ]: true,
-			} );
-		};
-
 		return (
 			<div className={ className }>
 				<BlockControls>
-					<ButtonGroup>
-						<button
-							className={ classesForPanel.call( this, 'editor' ) }
-							onClick={ this.showEditor }
-						>
-							<span>{ __( 'Markdown' ) }</span>
-						</button>
-						<button
-							className={ classesForPanel.call( this, 'preview' ) }
-							onClick={ this.showPreview }
-						>
-							<span>{ __( 'Preview' ) }</span>
-						</button>
-					</ButtonGroup>
+					<div className="components-toolbar">
+						{ this.renderToolbarButton( PANEL_EDITOR, __( 'Markdown' ) ) }
+						{ this.renderToolbarButton( PANEL_PREVIEW, __( 'Preview' ) ) }
```

Comment:
NIT Question: I'm starting to question render methods like these. This reduces some duplication, but also spreads the output so you don't have a good idea of what output to expect in one place.

Should they be Components? Is it even worth considering whether JSX or function calls should be used?

The following is a silly example that actually works just for consideration:

```diff
diff --git a/client/gutenberg/extensions/markdown/jetpack-markdown-block-editor.jsx b/client/gutenberg/extensions/markdown/jetpack-markdown-block-editor.jsx
index 78849cc394..3848be1871 100644
--- a/client/gutenberg/extensions/markdown/jetpack-markdown-block-editor.jsx
+++ b/client/gutenberg/extensions/markdown/jetpack-markdown-block-editor.jsx
@@ -33,7 +33,7 @@ class JetpackMarkdownBlockEditor extends Component {
 
 	toggleMode = mode => () => this.setState( { activePanel: mode } );
 
-	renderToolbarButton( mode, label ) {
+	ToolbarButton = ( { mode, label } ) => {
 		const { activePanel } = this.state;
 
 		return (
@@ -44,7 +44,7 @@ class JetpackMarkdownBlockEditor extends Component {
 				<span>{ label }</span>
 			</button>
 		);
-	}
+	};
 
 	render() {
 		const { attributes, className, isSelected } = this.props;
@@ -77,12 +77,14 @@ class JetpackMarkdownBlockEditor extends Component {
 			}
 		};
 
+		const This = this;
+
 		return (
 			<div className={ className }>
 				<BlockControls>
 					<div className="components-toolbar">
-						{ this.renderToolbarButton( PANEL_EDITOR, __( 'Markdown' ) ) }
-						{ this.renderToolbarButton( PANEL_PREVIEW, __( 'Preview' ) ) }
+						<This.ToolbarButton mode={ PANEL_EDITOR } label={ __( 'Markdown' ) } />
+						<This.ToolbarButton mode={ PANEL_PREVIEW } label={ __( 'Preview' ) } />
 					</div>
 				</BlockControls>
 				{ editorOrPreviewPanel.call( this ) }
```

Reply:
You're bringing a good point, I totally agree with that.

What stopped me from moving those toolbar buttons into separate buttons was our discussion previously: https://github.com/Automattic/wp-calypso/pull/27098#discussion_r216430822

I'm expecting that this will eventually change in Gutenberg, and then we'll use the Gutenberg components rather than introducing new ones just within this custom block.